### PR TITLE
deps: Run `yarn ugrade`

### PR DESCRIPTION
**Pull request by chrisbobbe**

### CODE REVIEW

**Comment by chrisbobbe** [comment]
If the `types` changes are a pain to review, perhaps I should redo the upgrade but hold back `react-native-safe-area-context`, which seems to be the source of most of the complexity there.

File: tools/tsflower [hunk 1]
```diff
@@ -209,9 +209,11 @@ run_only()
     rm -rf "${rootdir}"/types/"${package}"/lib/typescript/example/
     format_dir "${rootdir}"/types/"${package}"
 
-    # Remove all *.web.js.flow files. We don't use RN for the web.
+    # Remove all *.web.js.flow files, *.windows.js.flow files, etc.
```
Comment (gnprice) [inline tools/tsflower:212]: nit: can keep the "why" part of the comment:
```suggestion
    # Remove all *.web.js.flow files, etc. We don't use RN on those platforms.
```

**Review by gnprice — APPROVED** [review]
Thanks for taking care of these! This is a big bundle of upgrades we've accumulated; good to be taking them.

The types changes look good. (I agree that if we did need to split things out, holding back react-native-safe-area-context would be a good strategy.)

One nit below, then please merge at will.

**Comment by chrisbobbe** [comment]
Thanks for the review! Merging, with that tweak.

**Pull request merged**